File: src/lesson006/java/OverloadedMethods.java
package lesson006.java;

public class OverloadedMethods {
    public static void main(String[] args) {
        Overloaded overloaded = new Overloaded();
        System.out.println(overloaded.sumNumbers());
        System.out.println(overloaded.sumNumbers(1));
        System.out.println(overloaded.sumNumbers(1,2));
        System.out.println(overloaded.sumNumbers(1,2,3));
        System.out.println(overloaded.sumNumbers(1,2,3,4));
    }
}

class Overloaded{

    public int sumNumbers(){
        return 0;
    }

    public int sumNumbers(int arg1){
        return arg1;
    }

    public int sumNumbers(int arg1, int arg2){
        return arg1+arg2;
    }

    public int sumNumbers(int arg1, int arg2, int arg3){
        return arg1+arg2+arg3;
    }

    public int sumNumbers(int arg1, int arg2, int arg3, int arg4){
        return arg1+arg2+arg3+arg4;
    }
}

File: src/lesson006/kotlin/OverloadedMethods.kt
package lesson006.kotlin

fun main(args: Array<String>) {
    val over = Overloaded()
    println(over.sumNumbers())
    println(over.sumNumbers(1))
    println(over.sumNumbers(1,2))
    println(over.sumNumbers(1,2,3))
    println(over.sumNumbers(1,2,3,4))
}

class Overloaded() {
    fun sumNumbers() = 0
    fun sumNumbers(arg1: Int) = arg1
    fun sumNumbers(arg1: Int, arg2: Int) = arg1 + arg2
    fun sumNumbers(arg1: Int, arg2: Int, arg3: Int) = arg1 + arg2 + arg3
    fun sumNumbers(arg1: Int, arg2: Int, arg3: Int, arg4: Int) = arg1 + arg2 + arg3 + arg4
}